File: modules/foreground-service/android/src/main/java/com/ayush2006128/foreground/ForegroundServiceModule.kt
package com.ayush2006128.foreground

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ForegroundServiceModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ForegroundService")

    AsyncFunction("start") {
      ForegroundSensorService.start(requireNotNull(appContext.reactContext))
    }

    AsyncFunction("stop") {
      ForegroundSensorService.stop(requireNotNull(appContext.reactContext))
    }

    AsyncFunction("isRunning") {
      ForegroundSensorService.isRunning
    }
  }
}

File: modules/foreground-service/src/ForegroundServiceModule.ts
import { NativeModule, requireOptionalNativeModule } from "expo";
import type { ForegroundServiceModule as ForegroundServiceModuleType } from "./ForegroundService.types";

declare class ForegroundServiceModule extends NativeModule<{}> {
  start: ForegroundServiceModuleType["start"];
  stop: ForegroundServiceModuleType["stop"];
  isRunning: ForegroundServiceModuleType["isRunning"];
}

const nativeModule = requireOptionalNativeModule<ForegroundServiceModule>(
  "ForegroundService"
);

export default nativeModule ?? {
  start: async () => {},
  stop: async () => {},
  isRunning: async () => false,
};
